app/models/miq_worker.rb

```diff
@@ -346,8 +346,25 @@ def start_runner
     pid
   end
 
+  def self.build_command_line(guid)
+    command_line = "#{Gem.ruby} #{runner_script} --heartbeat --guid=#{guid} #{self.name}"
+    ENV['APPLIANCE'] ? "nice #{nice_increment} #{command_line}" : command_line
+  end
+
+  def self.runner_script
+    script = ManageIQ.root.join("lib/workers/bin/run_single_worker.rb")
+    raise "script not found: #{script}" unless File.exist?(script)
+    script
+  end
+
+  def start_runner_via_spawn
+    self.pid = Kernel.spawn(self.class.build_command_line(guid), :out => "/dev/null", :err => [Rails.root.join("log", "evm.log"), "a"])
+    Process.detach(pid)
+    pid
+  end
+
   def start
-    self.pid = start_runner
+    self.pid = ENV['MIQ_SPAWN_WORKERS'].to_s.downcase == "true" ? start_runner_via_spawn : start_runner_via_fork
```

Comment:
I didn't see you had omplemented it already, but you can drop the `.to_s.downcase == "true"` bit...it's not necessary